src/Query/AST/Functions/Postgresql/PostgresqlJsonFunctionNode.php

```diff
@@ -2,21 +2,21 @@
 
 namespace Scienta\DoctrineJsonFunctions\Query\AST\Functions\Postgresql;
 
-use Doctrine\DBAL\DBALException;
-use Doctrine\DBAL\Platforms\PostgreSqlPlatform;
+use Doctrine\DBAL\Exception;
+use Doctrine\DBAL\Platforms\PostgreSQL94Platform;
 use Doctrine\ORM\Query\SqlWalker;
 use Scienta\DoctrineJsonFunctions\Query\AST\Functions\AbstractJsonFunctionNode;
 
 abstract class PostgresqlJsonFunctionNode extends AbstractJsonFunctionNode
 {
     /**
      * @param SqlWalker $sqlWalker
-     * @throws DBALException
+     * @throws Exception
      */
     protected function validatePlatform(SqlWalker$sqlWalker): void
     {
-        if (!$sqlWalker->getConnection()->getDatabasePlatform() instanceof PostgreSqlPlatform) {
-            throw DBALException::notSupported(static::FUNCTION_NAME);
+        if (!$sqlWalker->getConnection()->getDatabasePlatform() instanceof PostgreSQL94Platform) {
```

Comment:
Json functions are available from postgres 9.3 and this limits future versions of postgres.
I would steer clear from such specific details and just leave this as it was: `PostgreSqlPlatform`

Reply:
This platform does not exist anymore. Th eparent of `PostgreSQL94Platform` is now `AbstractPlatform`.

Reply:
Doesnt look like it to me: https://github.com/doctrine/dbal/blob/3.2.x/src/Platforms/PostgreSQLPlatform.php

Reply:
And this: https://github.com/doctrine/dbal/blob/3.2.x/src/Platforms/PostgreSQL94Platform.php

Reply:
Hmm I'm confuse by their versioning. You're right for the 3.0 tag...

Reply:
When I look at the 3.2 tag it's there: https://github.com/doctrine/dbal/tree/3.2.1/src/Platforms

Reply:
Well, you're right about 3.0, but 3.2 of dbal deprecates the specific versions:
https://github.com/doctrine/dbal/blob/3.2.x/UPGRADE.md#deprecated-versioned-platform-classes-that-represent-the-lowest-supported-version

Reply:
It's a bit messy

Reply:
The, should I put ^3.2 for the version constraint?
They seem to have deprecated the Postgresql94Platform class in the 3.2 version.

Reply:
If you support doctrine 3.*, you should support both `PostgreSQLPlatform` and `PostgreSQL94Platform`.
(same for other platforms)

Reply:
You lost me here.
If we support 3.*, it means that 3.0 is supported. If 3.0 is supported, the `PostgreSQLPlatform` class does not exist and the code will crash.
`PostgreSQL94Platform` has been deprecated in 3.2 by the reintroduction of `PostgreSQLPlatform` and I expect it to be removed in 4.0.

I can replace `PostgreSQL94Platform` by `PostgreSQLPlatform`, but in this case, the version constraint becomes `3.*, >=3.2` or `^3.2` meaning all versions of the major release 3, greater or equal to 3.2.

Reply:
@piotaixr you're exactly right. Hence the messy comment...

Reply:
Can we resolve this, then?

Reply:
Can we resolve this, then?

Reply:
what's about

```suggestion
        $platform = !$sqlWalker->getConnection()->getDatabasePlatform();
        if (!$platform instanceof PostgreSQL94Platform && !$platform instanceof PostgreSQLPlatform) {
```

Reply:
Curious, I was expecting this to fail with doctrine 3.1. (Doctrine 3.0 cannot be used, constraints from doctrine/orm preventing it)
Now, what I don't know is if its possible for an instance of AbstractPlatform (not Postgresql) to be given to this function. If yes (and the API does not prevent it, so I think its possible), the code will break with somethong else than `Doctrine\DBAL\Exception` which violates the interface contract.

I don't understand why you are pushing this much to have the check also done on `PostgreSQLPlatform`, can you elaborate more?

Reply:
In the end we can deal with the deprecation later. I've just merged and released version 5. Thank you for all your effort.